src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/WebInputSources.cs

```diff
@@ -19,6 +21,21 @@ static WebInputSources()
         {
             var sourceInfosBuilder = PooledHashSet<SourceInfo>.GetInstance();
 
+            sourceInfosBuilder.AddSourceInfo(
+                WellKnownTypeNames.MicrosoftAspNetCoreMvcControllerBase,
+                 new ParameterMatcher[]{
+                    (parameter) => {
+                        ISymbol containingSymbol = parameter.ContainingSymbol;
+
```

Comment:
Definitely should take into account `[NonAction]` attributes, and possibly `[NonController]` attributes. We can't have the `INamedTypeSymbol`s at this point, so may have to do something like look at the `MetadataName` of `AttributeData.AttributeClass`, and walk up their `ContainingNamespace`s checking their `MetadataName`s.

https://docs.microsoft.com/en-us/aspnet/core/mvc/controllers/actions?view=aspnetcore-3.1 suggests that controllers don't necessarily have to derive from `ControllerBase`, but that's probably the most common case.

Reply:
Love the document, is there something like this for the old asp.net mvc?